components/srs_page/Flashcard.tsx

```diff
@@ -0,0 +1,104 @@
+import React, { useState, useEffect } from 'react';
+import { View, Text, TouchableOpacity } from 'react-native';
+import { Card } from '@/lib/superMemov2';
+import { Audio } from 'expo-av';
+import { Volume2 } from 'lucide-react-native';
+
+interface FlashcardProps {
+  card: Card;
+  onGrade: (card: Card, grade: number) => void;
+}
+
+const Flashcard: React.FC<FlashcardProps> = ({ card, onGrade }) => {
+  const [revealed, setRevealed] = useState(false);
+  const [sound, setSound] = useState<Audio.Sound | null>(null);
+
+  useEffect(() => {
+    setRevealed(false);
+    return () => {
+      if (sound) {
+        sound.unloadAsync();
+      }
+    };
+  }, [card]);
+
+  const playAudio = async (audioUrl: string) => {
+    if (sound) {
+      await sound.stopAsync();
+      sound.unloadAsync();
+    }
+
+    const { sound: newSound } = await Audio.Sound.createAsync({ uri: audioUrl });
+    setSound(newSound);
+    await newSound.playAsync();
+  };
+
+  return (
+    <View className="flex-1 w-full max-w-md justify-center items-center p-4">
+      <View className="bg-blue-100 p-6 rounded-lg shadow-md w-full">
+      <View className="flex flex-row justify-between items-center relative">
+          <View className="flex-1">
+            <Text className="text-3xl font-bold text-gray-800 text-center">{card.englishWord}</Text>
+          </View>
+          <TouchableOpacity onPress={() => playAudio(card.audioSource)} className="absolute right-0">
+            <Volume2 className="h-5 w-5 text-gray" />
+          </TouchableOpacity>
+        </View>
+        {revealed ? (
+          <Text className="text-2xl mt-4 text-center text-gray-700">{card.japaneseWord}</Text>
+        ) : (
+          <TouchableOpacity
+            onPress={() => setRevealed(true)}
+            className="mt-4 bg-blue-500 px-6 py-2 rounded-full"
+          >
+            <Text className="text-lg text-white text-center">Reveal</Text>
+          </TouchableOpacity>
+        )}
+      </View>
+
+      {revealed && (
+        <View className="mt-6 w-full">
```

Comment:
I think we could lose this first row and just score 1-3 / have 'easy', 'ok', 'hard'

Reply:
We should also add an option to completely remove the word from review e.g. 'mark as known'

Reply:
> I think we could lose this first row and just score 1-3 / have 'easy', 'ok', 'hard'

This is an important point and I'm still 50-50 on it.

The superMemo algorithm, which is basically being used here verbatim, is designed to work with 5 levels of retention. It's probably true that sacrificing some of that may be preferable from a user perspective.

In this case I modelled the buttons on the SRS app I use every day, [https://jpdb.io/](url). It uses five buttons in an identical layout. It seems to be becoming very popular and I find it more effective than alternatives (I got 100% in the language sections of my past two JLPT exams, which I attribte to it!) However, there is a good chance that the audience for JPDB is more 'hardcore' language learners than the average user who may want to try our app, and who might not appreciate the extra granularity. 

Maybe this is something we can user test down the line?

Reply:
I see! Personally I find 5 options just too much to think about which distracts from the actual task at hand. I understand that there might be a benefit if you don't mind, but for the more casual user I think it's too much. This is a product decision though so something we can definitely revisit when we have a clearer idea of what our product is.